utils/phased_barrier.hh

```diff
@@ -71,6 +71,16 @@ public:
         return old_gate->close().then([old_gate, op = start()] {});
     }
 
+    // Ends the current phase but doesn't start a new one. Any new attempts at
+    // starting an operation will result in `gate_closed` exception.
```

Comment:
`advance_and_await` above should check `is_closed()` and throw a `gate_closed` exception if it is.
Otherwise, calling `old_gate->close()` will hit an assert that it's already closed.

Reply:
Will do.